src/Serialization/main.jl

```diff
@@ -189,6 +193,14 @@ function save_type_params(s::SerializerState, obj::Any, key::Symbol)
   save_type_params(s, obj)
 end
 
+function save_attrs(s::SerializerState, obj::T) where T
+  save_data_dict(s, :attrs) do
+    for attr in attrs_list(s, T)
+      has_attribute(obj, attr) && save_typed_object(s, get_attribute(obj, attr), attr)
+    end
+  end
```

Comment:
this, currently, always saves a dict, which may often be empty. In particular, this also gets saved when passing `with_attrs=false`. To circumvent this, you could add something like the following:
```suggestion
  if any(attr -> has_attribute(obj, attr), attrs_list(s, T))
    save_data_dict(s, :attrs) do
      for attr in attrs_list(s, T)
        has_attribute(obj, attr) && save_typed_object(s, get_attribute(obj, attr), attr)
      end
    end
  end
```

Reply:
ill add the check here to not block lie algebra serialization, but also keep the check in toric varieties for now, and see if other similar examples popup and adjust them all at the same time.

Reply:
Great, thanks!